### PR TITLE
update golang to 1.19 and operator-sdk to 1.26

### PR DESCRIPTION
**Pull request by stuggi**
* update golang to 1.19 
* update operator-sdk to 1.26
* add revive to golangci
* update ginkgo to v2

### CODE REVIEW

File: controllers/openstackclient_controller.go [hunk 1]
```diff
@@ -586,8 +588,12 @@ func (r *OpenStackClientReconciler) podCreateOrUpdate(
...
 
 		if imageUpdate && isPodUpdate {
-			// init container image is mutable but does nothing so force a delete
-			return &common.ForbiddenPodSpecChangeError{Field: "Spec.InitContainers[0].Image"}
+			// init container image is mutable but does nothing so force a delete by triggering NewForbidden
+			return k8s_errors.NewForbidden(
+				schema.GroupResource{Group: "", Resource: "pods"}, // Specify the group and resource type
+				pod.Name,
+				errors.New("Cannot update Pod spec field - Spec.InitContainers[0].Image"), // Specify the error message
+			)
```
Comment (stuggi) [inline controllers/openstackclient_controller.go:596]: with new golangci config, it was complaining on the errors  type cast implementation in the now removed `pkg/common/errors.go`

File: api/v1beta1/webhook_suite_test.go [hunk 2]
```diff
@@ -1,12 +1,9 @@
 /*
-
-
+Copyright 2022.
```
Comment (abays) [inline api/v1beta1/webhook_suite_test.go:2]: I know this is auto-generated, but should we put 2023 here instead?

Comment (stuggi) [reply api/v1beta1/webhook_suite_test.go]: done

File: api/v1beta2/webhook_suite_test.go [hunk 3]
```diff
@@ -1,12 +1,9 @@
 /*
-
-
+Copyright 2022.
```
[the remark this answers is not in the document]
Comment (stuggi) [reply api/v1beta2/webhook_suite_test.go]: done

File: controllers/openstackclient_controller.go [hunk 4]
```diff
@@ -603,7 +609,8 @@ func (r *OpenStackClientReconciler) podCreateOrUpdate(
 		return nil
 	})
 	if err != nil {
-		if common.IsForbiddenPodSpecChangeError(err) || k8s_errors.IsInvalid(err) {
+		var forbiddenPodSpecChangeErr *k8s_errors.StatusError
+		if errors.As(err, &forbiddenPodSpecChangeErr) || k8s_errors.IsInvalid(err) {
```
Comment (olliewalsh) [inline controllers/openstackclient_controller.go:613]: will this be true for any status error, not just a Forbidden and Invalid?

Comment (stuggi) [reply controllers/openstackclient_controller.go:613]: right its on any status error, will update it

**Pull request merged**